### PR TITLE
{AKS} `az aks nodepool add/update/upgrade`: Add new parameter `--node-soak-duration`

### PR DESCRIPTION
**Pull request by MaxHorstmann**
**Related command**
az aks nodepool add|update|upgrade

**Description**
Bringing preview change https://github.com/Azure/azure-cli-extensions/pull/6865 to stable.

Allows customers to specify soak duration for a node during upgrades, i.e. the amount of time (in minutes) to wait after draining a node and before reimaging it and moving on to next node.

This was added to the 2023-11-01 API by https://github.com/Azure/azure-rest-api-specs/pull/26762. 

Note that this change is relatively similar to #27475.

**Testing Guide**

```
$ az aks create -g my-resource-group -n SoakTestCluster
$ az aks nodepool update -n nodepool1 -g my-resource-group --cluster-name SoakTestCluster --node-soak-duration 5
$ az aks upgrade -g my-resource-group -n SoakTestCluster -k 1.27.1 
$ kubectl get events | grep Soak 
```

**History Notes**
[AKS] `az aks nodepool add/update/upgrade`: Add new parameter `--node-soak-duration` to wait after draining a node and before reimaging it and moving on to next node

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -5869,6 +5869,45 @@ def test_aks_nodepool_drain_timeout(self, resource_group, resource_group_locatio
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    @AllowLargeResponse()
```
Comment (MaxHorstmann) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:5872]: TODO once Swagger change (https://github.com/Azure/azure-rest-api-specs/pull/26762) merged & SDK updated, queue live test and add recording. See earlier discussion on preview CLI with @FumingZhang [here](https://github.com/Azure/azure-cli-extensions/pull/6865/files#r1364821249)

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by zhoxing-ms** [comment]
@FumingZhang Could you please help review the AKS related PR?

**Comment by FumingZhang** [comment]
Need to wait for the update of SDK and default API version in aks command group, and it seems that the target milestone has been changed to December 2023.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (MaxHorstmann) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:5872]: Swagger change now merged, waiting for SDK update

Comment (MaxHorstmann) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:5893]: Looks like SDK release will happen shortly

https://github.com/Azure/azure-sdk-for-go/pull/22271

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 2]
```diff
@@ -32,7 +32,7 @@ azure-mgmt-cognitiveservices==13.5.0
 azure-mgmt-compute==30.4.0
 azure-mgmt-containerinstance==10.1.0
 azure-mgmt-containerregistry==10.1.0
-azure-mgmt-containerservice==28.0.0
+azure-mgmt-containerservice==29.0.0
```
Comment (MaxHorstmann) [inline src/azure-cli/requirements.py3.Darwin.txt:35]: https://pypi.org/project/azure-mgmt-containerservice/29.0.0

Python SDK with 2023-11-01 API, see https://github.com/Azure/sdk-release-request/issues/4847

Comment (MaxHorstmann) [reply src/azure-cli/requirements.py3.Darwin.txt:35]: @FumingZhang ok to just bump the SDK version here as part of this PR?

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (MaxHorstmann) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:5893]: Live test: https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=86189118&view=results

File: src/azure-cli/requirements.py3.Darwin.txt (hunk 2 above)
Comment (MaxHorstmann) [reply src/azure-cli/requirements.py3.Darwin.txt:35]: Apparently not, getting this [error](https://dev.azure.com/azclitools/public/_build/results?buildId=124273&view=logs&j=537f1e04-c6d5-53b2-3a1f-f7387cc6e02d&t=cae9a690-b596-5b1d-0d0e-9efe890cb6db&l=441):

```
The conflict is caused by:
    The user requested azure-mgmt-containerservice==29.0.0
    azure-cli 2.56.0 depends on azure-mgmt-containerservice~=28.0.0
```

Comment (FumingZhang) [reply src/azure-cli/requirements.py3.Darwin.txt:35]: I bumped the SDK in PR #28229. Please rebase from main branch to pick up the change.

Comment (MaxHorstmann) [reply src/azure-cli/requirements.py3.Darwin.txt]: Done, thank you!

**Review by FumingZhang — APPROVED** [review]
LGTM

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=86307573&view=results) to validate the change. Test passed!

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**